mw/i2c_service/data/header.h

```diff
@@ -0,0 +1,50 @@
+/**
+ * @file header.h
+ * @author Michal Mankowski (m.mankowski2004@gmail.com)
+ * @brief 
+ * @version 0.1
+ * @date 2024-03-14
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+
+#ifndef MW_I2C_SERVICE_DATA_HEADER_H_
+#define MW_I2C_SERVICE_DATA_HEADER_H_
+
+#include <cstdint>
+
+#include "communication-core/network-data/network_data_structure.h"
+#include "communication-core/network-data/network_data_type.h"
+
+
+namespace simba {
+namespace i2c {
+
+class Header : public com::core::network::NetworkDataStructure {
+ private:
+    com::core::network::uint8_t action;
+    com::core::network::uint8_t address;
+    com::core::network::uint8_t reg;
+    com::core::network::uint8_t service_id;
+    void SetData();
+ public:
+    enum ACTION {
+        Read = 0x01,
+        Write = 0x02,
+        ReadWrite = 0x03
+    };
+    ACTION GetAction() { return static_cast<ACTION>(this->action.Get()); }
+    uint8_t GetAddress() { return this->address.Get(); }
+    uint8_t GetReg() { return this->reg.Get(); }
+    uint8_t GetServiceId() { return this->service_id.Get(); }
```

Comment:
```suggestion
    ACTION GetAction() const { return static_cast<ACTION>(this->action.Get()); }
    uint8_t GetAddress() const { return this->address.Get(); }
    uint8_t GetReg() const { return this->reg.Get(); }
    uint8_t GetServiceId() const { return this->service_id.Get(); }
```

+ nie widze koniecznoci trzymania ich w pliku h